src/ripple/ledger/impl/View.cpp

```diff
@@ -348,9 +348,15 @@ xrpLiquid(
 
     auto const balance = view.balanceHook(id, xrpAccount(), fullBalance);
 
-    STAmount amount = balance - reserve;
-    if (balance < reserve)
-        amount.clear();
+    STAmount amount = [&]() {
+        // AMM doesn't require the reserves
+        if (sle->getFlags() & lsfAMM)
+            return balance;
```

Comment:
I understand why it's done, but I dislike exceptions like these. Exceptional cases like these are what make software complicated.